packages/cells/src/model.ts

```diff
@@ -612,9 +612,9 @@ export class CodeCellModel extends CellModel implements ICodeCellModel {
       if (cell && cell.cell_type === 'code') {
         executionCount.set(cell.execution_count || null);
         outputs = cell.outputs ?? [];
-        // If output is not empty presume it results of the input code execution
+        // If execution count is not null presume the input code was the latest executed
         // TODO load from the notebook file when the dirty state is stored in it
-        if (outputs.length > 0) {
+        if (cell.execution_count) {
```

Comment:
Minor nitpick: Maybe it should check for `null` as the specification only says it is an integer or null; 0 is a valid integer and I guess that an exotic kernel could start counting from 0. I would use `cell.execution_count != null` (not triple so that `undefined` is matched too; and I would tell eslint to ignore eqeqeq here).

Reply:
Great point. I'll update the PR.